### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by 9999years**
Split code out into files, add contracts

### CODE REVIEW

File: github.rkt [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+      (post-pure-port (github-review-request-url req)
+                      (jsexpr->bytes (github-review-request-jsexpr req))
+                      ; https://docs.github.com/en/rest/reference/pulls#list-review-comments-in-a-repository-preview-notices
+                      `("Accept: application/vnd.github.comfort-fade-preview+json"
+                        ,(format "Authorization: Bearer ~a" (github-token)))))
+    (define response-or-eof (read-json response-port))
+    (if (eof-object? response-or-eof)
+        (error (format "No response data for request to ~a"
+                       (github-review-request-url req)))
+        response-or-eof)))
```
Comment (jackfirth) [inline github.rkt]: nit: trailing newline

File: command.rkt [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+(require racket/contract)
+
+(provide (contract-out
+          [run-command (->* (string?) #:rest (listof (or/c string? path?))
+                            string?)]))
+
+(require racket/string
+         racket/port
+         rebellion/private/guarded-block)
```
Comment (jackfirth) [inline command.rkt]: nit: keep required libraries sorted, it makes it easier to deduplicate them and helps avoid merge conflicts.
```suggestion
(require racket/port
         racket/string
         rebellion/private/guarded-block)
```

File: command.rkt [hunk 3]
```diff
@@ -0,0 +1,40 @@
+#lang racket/base
+
+(require racket/contract)
```
Comment (jackfirth) [inline command.rkt]: nit: import `racket/contract/base` instead, it loads faster.
```suggestion
(require racket/contract/base)
```

File: command.rkt [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+; This doesn't support anything but getting stdout -- but that's OK for now!
+(define/guard (run-command cmd-name . args)
+  (define cmd-path (or (find-executable-path cmd-name)
+                       ; Racket doesn't know about $PATHEXT:
+                       (find-executable-path (string-append cmd-name ".exe"))))
+  (guard (not cmd-path) then
+         (error (format "couldn't find ~a executable in $PATH" cmd-name)))
+  (define-values (proc stdout stdin stderr)
+    (apply subprocess `(#f #f #f ,cmd-path ,@args)))
```
Comment (jackfirth) [inline command.rkt]: The `apply` function can take multiple arguments, and only the last one is used as a list. So you can simplify this a bit.
```suggestion
    (apply subprocess #f #f #f cmd-path args))
```

File: command.rkt [hunk 5]
```diff
@@ -0,0 +1,40 @@
+#lang racket/base
+
+(require racket/contract)
+
+(provide (contract-out
+          [run-command (->* (string?) #:rest (listof (or/c string? path?))
+                            string?)]))
```
Comment (jackfirth) [inline command.rkt]: You can use ellipses with the `->` contract combinator to specify rest arguments more easily.
```suggestion
          [run-command (-> string? (or/c string? path?) ... string?)]))
```

File: github.rkt (hunk 1 above)
Comment (9999years) [inline github.rkt]:
```suggestion
        response-or-eof)))

```

**Comment by 9999years** [comment]
Fixed nits, PTAL?

**Review by jackfirth — APPROVED** [review]

**Pull request merged**